File: my-folder/0792-binary-search/solution.ts
function search(nums: number[], target: number): number {
    if (nums.length < 1) return -1;

    return bs(0, nums.length - 1, nums, target);
};

function bs(i: number, j: number, nums: number[], target: number): number {
    if (i > j || nums[i] === undefined || nums[j] === undefined) return -1;
    if (i === j) {
        return (target === nums[i])? i : -1; 
    }

    const mid = Math.floor((i + j) / 2);

    if (target === nums[mid]) return mid;

    if (target > nums[mid]) {
        return bs(mid + 1, j, nums, target);
    } else {
        return bs(i, mid - 1, nums, target);
    }
    
}

File: my-folder/0792-binary-search/solution.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        return self.bin_search(nums, target, 0, len(nums) - 1)

    def bin_search(self, nums: List[int], target: int, start: int, end: int) -> int:
        mid = start + (end - start) // 2
        if start > end:
            return -1

        if nums[mid] == target:
            return mid
        elif nums[mid] > target:
            return self.bin_search(nums, target, start, mid - 1)
        else:
            return self.bin_search(nums, target, mid + 1, end)


        
